ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -1703,6 +1703,127 @@ class EclCmdStatus : public EclCmdCommon
+    {
+        optObj.accept = eclObjWuid;
+    }
+    virtual eclCmdOptionMatchIndicator parseCommandLineOptions(ArgvIterator &iter)
+    {
+        eclCmdOptionMatchIndicator retVal = EclCmdOptionNoMatch;
+        if (iter.done())
+            return EclCmdOptionNoMatch;
+
+        for (; !iter.done(); iter.next())
+        {
+            const char *arg = iter.query();
+            if (*arg!='-') //parameters don't start with '-'
+            {
+                if (optWuid.length())
+                {
+                    fprintf(stderr, "\nunrecognized argument %s\n", arg);
+                    return EclCmdOptionCompletion;
+                }
+                if (!looksLikeAWuid(arg, 'W'))
+                {
+                    fprintf(stderr, "\nargument should be a workunit id: %s\n", arg);
+                    return EclCmdOptionCompletion;
+                }
+                optWuid.set(arg);
+                continue;
+            }
+            if (iter.matchOption(optPath, ECLOPT_PATH))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            if (iter.matchFlag(optIncThorSlave, ECLOPT_INC_THOR_SLAVE_LOGS))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            if (iter.matchOption(optProblemDesc, ECLOPT_PROBLEM_DESC))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            eclCmdOptionMatchIndicator ind = EclCmdCommon::matchCommandLineOption(iter, true);
+            if (ind != EclCmdOptionMatch)
+                return ind;
+        }
+        return retVal;
+    }
+    virtual bool finalizeOptions(IProperties *globals)
+    {
+        if (!EclCmdCommon::finalizeOptions(globals))
+            return false;
+        return true;
+    }
+    virtual int processCMD()
+    {
+        Owned<IClientWsWorkunits> client = createCmdClient(WsWorkunits, *this);
+
+        Owned<IClientWUCreateZAPInfoRequest> zapReq = client->createWUCreateZAPInfoRequest();
+        zapReq->setWuid(optWuid.get());
+        zapReq->setIncludeThorSlaveLog(optIncThorSlave);
+        zapReq->setProblemDescription(optProblemDesc.get());
+
+        Owned<IClientWUCreateZAPInfoResponse> zapResp = client->WUCreateZAPInfo(zapReq);
+        int ret = outputMultiExceptionsEx(zapResp->getExceptions());
+        if (ret == 0)
+        {
+            StringBuffer filePath(optPath);
+            const char *p = filePath.str();
+            p = (p + strlen(p) - 1);
+            if (!streq(p, PATHSEPSTR))
+                filePath.append(PATHSEPSTR);
+
+            filePath.append(zapResp->getZAPFileName());
+
+            const MemoryBuffer & zapFileData = zapResp->getThefile();
+
+            Owned<IFile> zapFile = createIFile(filePath.str());
+            Owned<IFileIO> zapFileIo = zapFile->open(IFOcreate);
+            Owned<IFileIOStream> zapFileStream = createBufferedIOStream(zapFileIo);
```

Comment:
does IFileIOStream give you something in this case, rather than just calling?:

`zapFileIo->write(0, zapFileData.length(), zapFileData.toByteArray());`

Reply:
Not exactly, removed, changed.